### PR TITLE
More ASoC/cppcheck warnings

### PR DESCRIPTION
**Pull request by plbossart**
Two on my own code, the rest are trivial fixes.

### CODE REVIEW

File: sound/soc/soc-acpi.c [hunk 1]
```diff
@@ -34,7 +34,7 @@ static acpi_status snd_soc_acpi_find_package(acpi_handle handle, u32 level,
 					     void *context, void **ret)
 {
 	struct acpi_device *adev;
-	acpi_status status = AE_OK;
+	acpi_status status;
```
Comment (bardliao) [inline sound/soc/soc-acpi.c:37]: Not related to this patch, but should we return `status` in line 65?
```
		if (ACPI_FAILURE(status)) {
			kfree(buffer.pointer);
			return AE_OK;                        -> Should it be return status?
		}
```

Comment (lyakh) [reply sound/soc/soc-acpi.c:37]: ...and `kfree(buffer.pointer);` could be moved one line up from line 64 and removed from line 68 :-D But yes, unrelated

File: sound/soc/soc-core.c [hunk 2]
```diff
@@ -2780,12 +2780,12 @@ int snd_soc_of_parse_audio_routing(struct snd_soc_card *card,
 			propname);
 		return -EINVAL;
 	}
-	num_routes /= 2;
 	if (!num_routes) {
 		dev_err(card->dev, "ASoC: Property '%s's length is zero\n",
 			propname);
 		return -EINVAL;
 	}
+	num_routes /= 2;
```
Comment (lyakh) [inline sound/soc/soc-core.c:2783]: this one is actually strange... Isn't it a false positive? Assuming cppcheck actually only checks preprocessing. In fact if you look at `of_property_count_strings()` and then at `of_property_read_string_helper()` then indeed 0 is never returned there! For 0 `-ENODATA` is returned. But I wouldn't expect cppcheck to check that? But anyway wouldn't the best fix be to actually drop the `(!num_routes)` check?

File: sound/soc/codecs/rt5682.c [hunk 3]
```diff
@@ -2401,7 +2401,7 @@ static int rt5682_set_component_pll(struct snd_soc_component *component,
 			(pll_code.n_code << RT5682_PLL_N_SFT) | pll_code.k_code);
 		snd_soc_component_write(component, RT5682_PLL_CTRL_2,
 			((pll_code.m_bp ? 0 : pll_code.m_code) << RT5682_PLL_M_SFT) |
-			(pll_code.m_bp << RT5682_PLL_M_BP_SFT | RT5682_PLL_RST));
+			((pll_code.m_bp << RT5682_PLL_M_BP_SFT) | RT5682_PLL_RST));
```
Comment (lyakh) [inline sound/soc/codecs/rt5682.c:2404]: boolean??? cppcheck is definitely less than perfect...

**Review by lyakh — CHANGES_REQUESTED** [review]

**Review by RanderWang — APPROVED** [review]
LGTM

File: sound/soc/soc-core.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/soc-core.c:2783]: < 0 is an error
odd values are not possible, so 1 is not allowed

at this point it's either 0, or 2+

What cppcheck detected is that if num_routes = 2+, then after the division by two num_routes cannot be zero.

if zero cannot happen, then we can remove this condition, but that's not something cppcheck would verify on its own.

File: sound/soc/soc-acpi.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/soc-acpi.c:37]: > Not related to this patch, but should we return `status` in line 65?
> 
> ```
> 		if (ACPI_FAILURE(status)) {
> 			kfree(buffer.pointer);
> 			return AE_OK;                        -> Should it be return status?
> 		}
> ```

the AE_OK is actually an error value that will prompt the ACPI code to move to the next element. We always return AE_OK, except for the case where we return AE_TERMINATE which essentially means success.

don't ask me why it was designed this way, this is probably 30 years old now ;-)

Comment (plbossart) [reply sound/soc/soc-acpi.c:37]: > ...and `kfree(buffer.pointer);` could be moved one line up from line 64 and removed from line 68 :-D But yes, unrelated

yes, but in this case I prefer to leave the code as is. It was literally copy/pasted/adjusted from other sources, it's better to keep the same 'style' IMHO.

File: sound/soc/soc-core.c (hunk 2 above)
Comment (lyakh) [reply sound/soc/soc-core.c:2783]: cppcheck complained that "Condition '!num_routes' is always false" and that's exactly what surprised me - that seems to suggest, that it recognised, that `num_routes` cannot be 0, for which it had to check the `of_property_count_strings()` function which would be a strange thing to do for a static code analyser...

Comment (plbossart) [reply sound/soc/soc-core.c:2783]: I removed this test as suggested, it cannot happen. Thanks for the feedback and detailed reviews, much appreciated @lyakh

File: sound/soc/codecs/rt5682.c (hunk 3 above)
Comment (plbossart) [reply sound/soc/codecs/rt5682.c:2404]: it's still the best tool so far in my experience.

**Review by dbaluta — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Pull request merged**